android/expoview/src/main/java/host/exp/exponent/experience/ExperienceActivity.java

```diff
@@ -388,30 +405,24 @@ protected void showOrReconfigureManagedAppSplashScreen(final JSONObject manifest
     }
   }
 
-  private void maybeSetLoadingProgressStatus() {
+  public void maybeSetLoadingProgressStatus() {
     ExpoUpdatesAppLoader appLoader = mKernel.getAppLoaderForManifestUrl(mManifestUrl);
     if (appLoader != null) {
-      setLoadingProgressStatus(appLoader.getStatus());
+      maybeSetLoadingProgressStatus(appLoader.getStatus());
     }
   }
 
-  private String getLoadingProgressText(ExpoUpdatesAppLoader.AppLoaderStatus status) {
-    if (status == ExpoUpdatesAppLoader.AppLoaderStatus.CHECKING_FOR_UPDATE) {
-      return "Checking for new release...";
-    } else if (status == ExpoUpdatesAppLoader.AppLoaderStatus.DOWNLOADING_NEW_UPDATE) {
-      return "New release available, downloading...";
-    }
-    return null;
-  }
-
-  public void setLoadingProgressStatus(ExpoUpdatesAppLoader.AppLoaderStatus status) {
+  public void maybeSetLoadingProgressStatus(ExpoUpdatesAppLoader.AppLoaderStatus status) {
```

Comment:
A couple of thoughts:

- Would `setLoadingProgressStatusIfEnabled` be a slightly more descriptive name?
- Do you think there is merit in always setting the loading progress status but displaying it only when enabled?